_includes/manuals/1.24/5.7.7_openid_sso.md

```diff
@@ -0,0 +1,106 @@
+### Single Sign-on for Foreman using OpenID Connect protocol
+
+An OpenID Provider(OP)  implements a Single Sign-on (SSO) using  an OpenID Connect (OIDC) protocol. OIDC verifies the identity of the End-user depending on the given authentication details. In this example, [Keycloak](https://www.keycloak.org/) is used as the OpenID provider, and Foreman is the Relying Party (RP)
+
+OpenID provider provides an ID Token, encoded in JSON Web Token (JWT) to Foreman.
+
+An example of an encoded JWT ID Token:
+
+```
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
+```
+
+#### Configuring Foreman
+
+As a system administrator, execute the following procedure to configure Foreman.
+
+Prerequisites
+
+* A working installation of Foreman at https://foreman.example.com.
+* A working installation of OpenID provider, for example, Keycloak at https://sso.example.com, which uses OIDC protocol. Ensure that the Keycloak server is running.
+* Ensure that both RP and OpenID provider are using https instead of http. If the certificates or the CA is self-signed, ensure that they are added to the End-user certificate trust store.
+
+Procedure
+
+1. Install the mod_auth packages:
+```
+# yum install mod_auth_openidc python2-requests-oauthlib keycloak-httpd-client-install -y
+```
+
+2. Register Katello/Foreman as application in the Keycloak:
+```
+# export KEYCLOAK_URL=https://keycloak.example.com
+# export KEYCLOAK_REALM=ssl-realm
+# export KEYCLOAK_USER=admin
+# keycloak-httpd-client-install --app-name foreman-openidc --keycloak-server-url $KEYCLOAK_URL --keycloak-admin-username $KEYCLOAK_USER --keycloak-realm $KEYCLOAK_REALM --keycloak-admin-realm master --keycloak-auth-role root-admin -t openidc -l /users/extlogin -d
```

Comment:
This appears to fail for me. First I had to install python-lxml:
```
[root@68e8ec694948 /]# keycloak-httpd-client-install --help
Traceback (most recent call last):
  File "/usr/bin/keycloak-httpd-client-install", line 8, in <module>
    from lxml import etree
ImportError: No module named lxml
```

After that it doesn't recognize `-t`

```
[root@68e8ec694948 /]# export KEYCLOAK_URL=https://keycloak.example.com
[root@68e8ec694948 /]# export KEYCLOAK_REALM=ssl-realm
[root@68e8ec694948 /]# export KEYCLOAK_USER=admin
[root@68e8ec694948 /]# keycloak-httpd-client-install --app-name foreman-openidc --keycloak-server-url $KEYCLOAK_URL --keycloak-admin-username $KEYCLOAK_USER --keycloak-realm $KEYCLOAK_REALM --keycloak-admin-realm master --keycloak-auth-role root-admin -t openidc -l /users/extlogin -d
usage: keycloak-httpd-client-install [-h] [--no-root-check] [-v] [-d]
                                     [--show-traceback] [--log-file LOG_FILE]
                                     --app-name APP_NAME [--force]
                                     [--permit-insecure-transport]
                                     [--tls-verify TLS_VERIFY]
                                     [--template-dir TEMPLATE_DIR]
                                     [--httpd-dir HTTPD_DIR] -r KEYCLOAK_REALM
                                     -s KEYCLOAK_SERVER_URL
                                     [-a {root-admin,realm-admin,anonymous}]
                                     [-u KEYCLOAK_ADMIN_USERNAME]
                                     [-P KEYCLOAK_ADMIN_PASSWORD_FILE]
                                     [-p DEPRECATED_KEYCLOAK_ADMIN_PASSWORD]
                                     [--keycloak-admin-realm KEYCLOAK_ADMIN_REALM]
                                     [--initial-access-token INITIAL_ACCESS_TOKEN]
                                     [--client-originate-method {descriptor,registration}]
                                     [--mellon-key-file MELLON_KEY_FILE]
                                     [--mellon-cert-file MELLON_CERT_FILE]
                                     [--mellon-hostname MELLON_HOSTNAME]
                                     [--mellon-https-port MELLON_HTTPS_PORT]
                                     [--mellon-root MELLON_ROOT]
                                     [--mellon-endpoint MELLON_ENDPOINT]
                                     [--mellon-entity-id MELLON_ENTITY_ID]
                                     [--mellon-idp-attr-name MELLON_IDP_ATTR_NAME]
                                     [--mellon-organization-name MELLON_ORGANIZATION_NAME]
                                     [--mellon-organization-display-name MELLON_ORGANIZATION_DISPLAY_NAME]
                                     [--mellon-organization-url MELLON_ORGANIZATION_URL]
                                     [-l MELLON_PROTECTED_LOCATIONS]
keycloak-httpd-client-install: error: unrecognized arguments: -t openidc
```

Reply:
@ekohl  which version of `mod_auth_openidc` and `keycloak-httpd-client-install`
 are you using?

Reply:
Looks like currently CentOS ships the following:
* keycloak-httpd-client-install.noarch 0:0.8-1.el7
* mod_auth_openidc.x86_64 0:1.8.8-5.el7

Reply:
keycloak-httpd-client-install.noarch version `0:0.8-1.el7` could be the reason for failure. I reckon OpenID Connect support was added to keycloak-httpd-client-install in version 1.x. 

@rahulbajaj0509 Did you test this setup with a custom built keycloak-httpd-client-install?
@upadhyeammit FYI

Reply:
@gnurag that is correct, keycloak-httpd-client-install package with version >= 1.x. supports OpenID Connect.

@ekohl can we have this bit merged for now, so that we have some documentation about the feature rather than nothing? and then we can make improvements to it as when we face issues.
I am completely ok to not merge this if you believe so.

Reply:
I would add a note that you need at least version 1.x and the one in base is too old.

Reply:
Makes sense.

Reply:
I have added a note for keycloak-httpd-client-install package version.